Rule `no-unexpected-multiline` does not detect case which line starts with `/`

<!--
This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes
3. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**Tell us about your environment**

* **ESLint Version:**
v3.19.0
* **Node Version:**
v6.10.1
* **npm Version:**
3.10.10
**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

<!-- Paste your configuration below: -->
```
{
    "env": {
        "browser": true,
        "es6": true
    },
    "extends": "eslint:recommended",
    "rules": {
        "indent": [
            "error",
            "tab"
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "error",
            "double"
        ],
        "semi": [
            "error",
            "never"
        ],
        "no-console": [
            "off"
        ]
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

<!-- Paste the source code below: -->
```js
var g = 1

[g].forEach(_ => _)

g = 2

`g`

var i = 3

/g/i.test("G") && console.log(g)
```

**What did you expect to happen?**
Three errors are detected as `no-unexpected-multiline`

- Before `g = 2`
- Before ``` `g` ```
- Before `/g/i.test("G") && console.log(g)`

**What actually happened? Please include the actual, raw output from ESLint.**
Only two errors were detected.

```
  3:1  error  Unexpected newline between object and [ of property access    no-unexpected-multiline
  5:5  error  Unexpected newline between template tag and template literal  no-unexpected-multiline

✖ 2 problems (2 errors, 0 warnings)
```

--- a/docs/rules/no-unexpected-multiline.md
+++ b/docs/rules/no-unexpected-multiline.md
@@ -32,6 +32,9 @@ let x = function() {}
 let x = function() {}
 x
 `hello`
+
+let x = foo
+/regex/g.test(bar)
 ```
 
 Examples of **correct** code for this rule:

--- a/lib/rules/no-unexpected-multiline.js
+++ b/lib/rules/no-unexpected-multiline.js
@@ -30,6 +30,9 @@ module.exports = {
         const FUNCTION_MESSAGE = "Unexpected newline between function and ( of function call.";
         const PROPERTY_MESSAGE = "Unexpected newline between object and [ of property access.";
         const TAGGED_TEMPLATE_MESSAGE = "Unexpected newline between template tag and template literal.";
+        const DIVISION_MESSAGE = "Unexpected newline between numerator and division operator.";
+
+        const REGEX_FLAG_MATCHER = /^[gimuy]+$/;
 
         const sourceCode = context.getSourceCode();
 
@@ -75,6 +78,19 @@ module.exports = {
                     return;
                 }
                 checkForBreakAfter(node.callee, FUNCTION_MESSAGE);
+            },
+
+            "BinaryExpression[operator='/'] > BinaryExpression[operator='/']"(node) {
+                const secondSlash = sourceCode.getTokenAfter(node, token => token.value === "/");
+                const tokenAfterOperator = sourceCode.getTokenAfter(secondSlash);
+
+                if (
+                    tokenAfterOperator.type === "Identifier" &&
+                    REGEX_FLAG_MATCHER.test(tokenAfterOperator.value) &&
+                    secondSlash.range[1] === tokenAfterOperator.range[0]
+                ) {
+                    checkForBreakAfter(node.left, DIVISION_MESSAGE);
+                }
             }
         };
 

--- a/tests/lib/rules/no-unexpected-multiline.js
+++ b/tests/lib/rules/no-unexpected-multiline.js
@@ -200,17 +200,6 @@ ruleTester.run("no-unexpected-multiline", rule, {
                 column: 17,
                 message: "Unexpected newline between numerator and division operator."
             }]
-        },
-        {
-            code: `
-                foo
-                /bar/gimuygimuygimuy.test(baz)
-            `,
-            errors: [{
-                line: 3,
-                column: 17,
-                message: "Unexpected newline between numerator and division operator."
-            }]
         }
     ]
 });

--- a/tests/lib/rules/no-unexpected-multiline.js
+++ b/tests/lib/rules/no-unexpected-multiline.js
@@ -42,7 +42,43 @@ ruleTester.run("no-unexpected-multiline", rule, {
         {
             code: "x\n.y\nz `Valid Test Case`",
             parserOptions: { ecmaVersion: 6 }
-        }
+        },
+        `
+            foo
+            / bar /2
+        `,
+        `
+            foo
+            / bar / mgy
+        `,
+        `
+            foo
+            / bar /
+            gym
+        `,
+        `
+            foo
+            / bar
+            / ygm
+        `,
+        `
+            foo
+            / bar /GYM
+        `,
+        `
+            foo
+            / bar / baz
+        `,
+        "foo /bar/g",
+        `
+            foo
+            /denominator/
+            2
+        `,
+        `
+            foo
+            / /abc/
+        `
     ],
     invalid: [
         {
@@ -119,6 +155,61 @@ ruleTester.run("no-unexpected-multiline", rule, {
                 line: 3,
                 column: 1,
                 message: "Unexpected newline between template tag and template literal."
+            }]
+        },
+        {
+            code: `
+                foo
+                / bar /gym
+            `,
+            errors: [{
+                line: 3,
+                column: 17,
+                message: "Unexpected newline between numerator and division operator."
+            }]
+        },
+        {
+            code: `
+                foo
+                / bar /g
+            `,
+            errors: [{
+                line: 3,
+                column: 17,
+                message: "Unexpected newline between numerator and division operator."
+            }]
+        },
+        {
+            code: `
+                foo
+                / bar /g.test(baz)
+            `,
+            errors: [{
+                line: 3,
+                column: 17,
+                message: "Unexpected newline between numerator and division operator."
+            }]
+        },
+        {
+            code: `
+                foo
+                /bar/gimuygimuygimuy.test(baz)
+            `,
+            errors: [{
+                line: 3,
+                column: 17,
+                message: "Unexpected newline between numerator and division operator."
+            }]
+        },
+        {
+            code: `
+                foo
+                /bar/gimuygimuygimuy.test(baz)
+            `,
+            errors: [{
+                line: 3,
+                column: 17,
+                message: "Unexpected newline between numerator and division operator."
             }]
         }
     ]

